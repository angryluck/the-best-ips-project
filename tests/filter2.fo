fun int write_int(int x) = write(x)

fun [int] write_int_arr([int] x) = map(write_int, x)

fun bool isMul16(int a) =
  (a / 16) * 16 == a

fun [int] process(int n) =
  let x = filter(fn bool (int a) => a == (a / 2) * 2, iota(n)) in
  let y = map(fn int (int a) => a * a, x) in
  filter(isMul16, y)

fun int main() =
  let n1 = read(int) in
  let tmp1 = write_int_arr(process(n1)) in

  let n2 = read(int) in
  let tmp2 = write_int_arr(process(n2)) in

  let n3 = read(int) in
  let tmp3 = write_int_arr(process(n3)) in

  let n4 = read(int) in
  let tmp4 = write_int_arr(process(n4)) in

  let n5 = read(int) in
  let tmp5 = write_int_arr(process(n5)) in
  0
