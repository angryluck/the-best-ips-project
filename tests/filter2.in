20
10
1
33
17
