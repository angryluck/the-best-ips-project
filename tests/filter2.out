0
16
64
144
256
0
16
64
0
0
16
64
144
256
400
576
784
1024
0
16
64
144
256
